web/pdf_print_service.js

```diff
@@ -200,7 +211,17 @@
       if (!activeService) {
         console.error('Expected print service to be initialized.');
       }
-      activeService.renderPages().then(startPrint, abort);
+      var activeServiceOnEntry = activeService;
+      activeService.renderPages().then(function() {
+        activeServiceOnEntry.throwIfInactive();
+        return startPrint(activeServiceOnEntry);
+      }).catch(function() {
+        // Ignore any error messages.
+      }).then(function() {
+        if (activeServiceOnEntry.active) {
+          abort();
```

Comment:
So `activeServiceOnEntry == activeSerice` and we are calling abort() which cancels activeService a.k.a. activeServiceOnEntry -- that's still confuses me. Can we change the code so it will be read better, e.g. move into the startTask method or add more readable check (not sure if more comments are suitable here)?